### ISSUE TITLE
Very very basic SCP support

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2594]
|[<img alt="philmayers's avatar" src="https://avatars.githubusercontent.com/u/739238?s=50" width="50" height="50">](https://github.com/philmayers)| @philmayers reported|
|-|-|
|Trac ID|trac#2594|
|Type|enhancement|
|Created|2007-04-23 09:54:38Z|

The following is a basic SCP client for conch. It is only capable of getting a single file and there are no tests (I'll try to follow up with something better). It might be useful to someone.

Attachments:

* [tscp](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/535/535b58b6f0d309b567c637d3b61997efbefd6dd4/347350226bcc76f8531070c67dd646bb04ff3c8d) (8026 bytes) - added by philmayers on 2007-04-23 09:56:25Z - Very very basic SCP/SFTP client

<details><summary>Searchable metadata</summary>

```
trac-id__2594 2594
type__enhancement enhancement
reporter__philmayers philmayers
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__conch conch
keywords__ 
time__1177322078000000 1177322078000000
changetime__1265646099000000 1265646099000000
version__None None
owner__jesstess jesstess
cc__jesstess cc__khorn
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="philmayers's avatar" src="https://avatars.githubusercontent.com/u/739238?s=50" width="50" height="50">](https://github.com/philmayers)<a name="note_1"></a>|@philmayers commented|
|-|-|

Note, the client is SCP/SFTP capable. The SFTP component has the ability to "tail" a file over SFTP. This might also be useful to someone.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess commented|
|-|-|

If people want SCP support in Twisted I'm willing to spearhead addressing review comments on this patch and adding tests.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @jesstess|
|-|-|

I'm not sure there's much reason to support the old SCP protocol.  It's very bad, and a lot of the world has transitioned to the SFTP-based SCP tools.  Conch has SFTP support now, but maybe there could be some higher-level APIs.  There's probably some code relevant to this end in `twisted.vfs`, which hasn't seem much attention in a while (and I'm not sure that it's worth giving it any attention, aside from ripping nice stuff out of it and then throwing the rest away).

I suspect that documenting Conch's existing APIs, continuing to clean up Conch's test suite, and generally making the connection setup APIs (the ones that apply to anything you want to do with SSH, not anything that's SFTP specific) easier to use would be a more useful focus, though.

If we had a "lack of interest" resolution type, I would probably resolve this ticket as that.  I don't feel strongly enough to say "invalid".  Maybe "wontfix" is close enough, though.

**Comment by twisted-trac** [comment]
|[<img alt="philmayers's avatar" src="https://avatars.githubusercontent.com/u/739238?s=50" width="50" height="50">](https://github.com/philmayers)<a name="note_6"></a>|@philmayers commented|
|-|-|

I originally wrote the code for talking to Cisco routers and Netscreen firewalls which either do not or did not at the time support SFTP. However, I then discovered that Conch has other problems (namely, the lack of the full suite of SSH protocol bug workarounds that clients like OpenSSH have) which prevent those applications being really useful in the real world.

I also have little interest in doing UQDS work, which is why I abandoned it.

In short: bin it as far as I'm concerned. I had forgotten the ticket even existed.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_7"></a>|@jesstess set status to `closed`|
|-|-|

Thanks for the response, philmayers.
